### PR TITLE
refactoring: refactor code by standards

### PR DESCRIPTION
**Pull request by alexDose**

### CODE REVIEW

File: src/assets/icons/AvatarIcon.tsx [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
+}: {
+  width: string;
+  height: string;
+}) => (
   <svg
-    width='38'
-    height='37'
+    width={width}
+    height={height}
     viewBox='0 0 38 37'
```
Comment (gitkoshelew) [inline src/assets/icons/AvatarIcon.tsx:11]: you can use variables for viewBox also

File: src/components/layout/header/HeaderAccount.tsx [hunk 2]
```diff
@@ -1,38 +1,30 @@
...
-        <Link
-          textOverflow='ellipsis'
-          overflow='hidden'
-          whiteSpace='nowrap'
+  if (profileData.name) {
+    return (
+      <>
+        <Link href='/profilePage'>
+          <Avatar
+            sx={{ width: 38, height: 37 }}
```
Comment (gitkoshelew) [inline src/components/layout/header/HeaderAccount.tsx]: Use variables as in AvatarIcon

**Review by gitkoshelew — CHANGES_REQUESTED** [review]

File: src/assets/icons/AvatarIcon.tsx (hunk 1 above)
Comment (alexDose) [reply src/assets/icons/AvatarIcon.tsx:11]: if i use variables for viewBox then dimensions(width and height) don't work

Comment (alexDose) [reply src/assets/icons/AvatarIcon.tsx:11]: i tried to write viewBox={`0 0 ${width} ${height}`}. it don't work

File: src/components/layout/header/HeaderAccount.tsx (hunk 2 above)
Comment (alexDose) [reply src/components/layout/header/HeaderAccount.tsx]: AvatarIcon is svg and Avatar is MUI component. I don't use variables in Avatar as in svg. Maybe i don't understood.

File: src/components/layout/header/HeaderAccount.tsx [hunk 3]
```diff
@@ -1,38 +1,30 @@
...
-        />
-      ) : (
-        <AvatarIcon />
-      )}
-      {profileData.nickname ? (
-        <Link
-          textOverflow='ellipsis'
-          overflow='hidden'
-          whiteSpace='nowrap'
+  if (profileData.name) {
```
Comment (AnvarAka1) [inline src/components/layout/header/HeaderAccount.tsx]: You are checking for profileData.name, but you are using profileData.nickname inside. That is confusing.

When a person reads this code, they think that profileData.name is directly related to the following JSX code, which is not true. 
Should be if(profileData.nickname)

File: src/components/layout/navigation/Category.tsx [hunk 4]
```diff
@@ -34,10 +34,8 @@ export const Category = ({ categories }: CategoryType) => {
         scrollButtons='auto'
         aria-label='scrollable auto tabs example'
       >
-        {t<string, string[]>('categories', {
-          returnObjects: true,
-        }).map((categoryName, index) => (
-          <Tab label={categoryName} key={index} />
+        {categories.map((category) => (
+          <Tab label={t(category)} key={category} />
```
Comment (AnvarAka1) [inline src/components/layout/navigation/Category.tsx]: Create custom component named Tab, in which there will be useTranslation hook called.
You need to abstract the translation by creating custom components, which under the hood translate your texts

File: src/assets/icons/AvatarIcon.tsx (hunk 1 above)
Comment (gitkoshelew) [reply src/assets/icons/AvatarIcon.tsx:11]: should be wrapped to \`  , not only {}, i think

Comment (alexDose) [reply src/assets/icons/AvatarIcon.tsx:11]: sorry, i mean viewBox={`0 0 ${width} ${height}`}

Comment (alexDose) [reply src/assets/icons/AvatarIcon.tsx:11]: i can't write ` here!)

Comment (alexDose) [reply src/assets/icons/AvatarIcon.tsx:11]: i wrapped to ` it

**Review by gitkoshelew — APPROVED** [review]

**Review by visiteur85 — APPROVED** [review]

**Pull request merged**